### PR TITLE
[IMP] account: add total amount between tax_group name and tax amount if multi invoice line and multi tax group in SO/invoice report

### PR DESCRIPTION
**Pull request by dpa-odoo**
Description of the issue/feature this PR addresses:
- We need a details of taxes when we sell goods so customer know detailed amount of tax.

Current behavior before PR:
- Currently no any amount displayed when multi invoice line and multi tax groups in SO/invoice report.

Desired behavior after PR is merged:
- Now amount is display in () when multi invoice line and multi tax groups in SO/invoice report.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

Task : https://www.odoo.com/web#id=30501&view_type=form&model=project.task&action=327&menu_id=4720

Pad : https://pad.odoo.com/p/r.4c5e2a8c432e737fde588c32fdf2e769

### CODE REVIEW

File: addons/account/models/account_invoice.py [hunk 1]
```diff
@@ -1120,13 +1120,25 @@ def _track_subtype(self, init_values):
...
+        vals = {}
         currency = self.currency_id or self.company_id.currency_id
         for line in self.tax_line_ids:
             res.setdefault(line.tax_id.tax_group_id, 0.0)
             res[line.tax_id.tax_group_id] += line.amount
-        res = sorted(res.items(), key=lambda l: l[0].sequence)
-        res = map(lambda l: (l[0].name, l[1]), res)
-        return res
+        for inv_line in self.invoice_line_ids:
+            for tax_group in inv_line.invoice_line_tax_ids.mapped('tax_group_id'):
```
Comment (mdi-odoo) [inline addons/account/models/account_invoice.py]: No need to iterate over **invoice_line_ids** to find the **base** amount on which tax is applied. You can use **base** amount from **tax_line_ids** .

File: addons/sale/models/sale.py [hunk 2]
```diff
@@ -510,9 +512,19 @@ def _get_tax_amount_by_group(self):
                 res[group] += amount
                 if tax.include_base_amount:
                     base_tax += amount
-        res = sorted(res.items(), key=lambda l: l[0].sequence)
-        res = map(lambda l: (l[0].name, l[1]), res)
-        return res
+        for line in self.order_line:
```
Comment (mdi-odoo) [inline addons/sale/models/sale.py]: Same as above. May be you can use **base** amount .

File: addons/sale/models/sale.py [hunk 3]
```diff
@@ -510,9 +512,18 @@ def _get_tax_amount_by_group(self):
                 res[group] += amount
                 if tax.include_base_amount:
                     base_tax += amount
-        res = sorted(res.items(), key=lambda l: l[0].sequence)
-        res = map(lambda l: (l[0].name, l[1]), res)
-        return res
+            for tax_group in line.tax_id.mapped('tax_group_id'):
+                if tax_group in res.keys():
```
Comment (mdi-odoo) [inline addons/sale/models/sale.py]: Same as above. May be you can use **base** amount .

**Comment by hmo-odoo** [comment]
continue work on PR: https://github.com/odoo/odoo/pull/16724

**Pull request closed without merging**